runtime/java/src/common/RTTI.java

```diff
@@ -0,0 +1,41 @@
+package common;
+
+import java.util.*;
+
+abstract public class RTTI {
```

Comment:
Needs a doc comment.  

Also, it might not be immediately obvious that RTTI is short for "runtime type information".  Not sure that is even the best name since we already do have some runtime type info infrastructure, that is unrelated to this.  Maybe even just move this to be static members/methods of the `Prodleton`/`Terminalton` classes?